src/repo-url.js

```diff
@@ -0,0 +1,54 @@
+const fs = require('fs-extra');
+const path = require('path');
+const shared = require('./utils/shared');
+
+const preload = (content, resourcePath) => {
+  if (!resourcePath.match(/app-extras\.module\.ts$/)) {
+    return content;
+  }
+
+  function generateRepoUrlObject(url) {
+    const githubFilePathRoot = '/tree/master/src/app/';
+    const vstsFilePathRoot = '?path=%2Fsrc%2Fapp%2F';
+    const vstsRepoBranchSelector = '&version=GBmaster';
+
+    let result = {};
+
+    if (url.includes('visualstudio')) {
+      result.url = url + vstsFilePathRoot;
+      result.vstsBranchSelector = vstsRepoBranchSelector;
+    } else {
+      result.url = url + githubFilePathRoot;
+    }
+
+    return result;
+  }
+
+  let packageJson;
+
+  try {
+    packageJson = fs.readJsonSync(path.join(process.cwd(), 'package.json'));
+  } catch (error) {
+    throw new shared.StachePluginError(error.message);
+  }
+
+  let url = packageJson.repository.url;
```

Comment:
So what happens if they do not have a specified repository url yet?  Should we handle this or just let it return a broken github link?

Reply:
That key is always there, but if they're trying to add an edit in github button without there being a repository, that' on them. I can throw a warning, but I don't think there's much else we can do to handle it.

Reply:
Got ya, so the button is opt-in?

Reply:
Yeah. You have to set `showEdit: true` in your `skyuxconfig.json` in order for the button to show up at all.

Reply:
Or manually enable it on your `stache` tag for an individual page.

Reply:
Though this plugin will run regardless of that setting?  Should we short-circuit this if that is false or non existent?

`!resourcePath.match(/app-extras\.module\.ts$/) || !skyConfig.app.showEdit` or something? 

If for some reason the repository key is missing in the `package.json`, this plugin will blow up, It may be worth adding the default `''` to the key here. 
```
let url = '';
if(packageJson.repository && packageJson.repository.url) {
   url = packageJson.repository.url;
}
```
or something along those lines.

Reply:
I'm fine short circuiting it if that setting isn't set to true. That makes sense to save us a step. I don't really see the point in doing the check for the url if we do the short circuit. If you run `skyux new` that key is there, always. So unless they actively delete it, it'll be there. I'd argue we'd want the plugin to blow up if they've turned on the `showEdit` but there's not a repository object in the `package.json`